### ISSUE TITLE
nightly-2016-12-16 compiling Hello World on i686: `assertion failed: eps.windows(2).all(|w| w[0].cmp(self, &w[1]) != Ordering::Greater)`

### ISSUE THREAD

**Issue by i336** [rust-lang/rust#38898]
Just tried to get Rust working on a 32-bit system and a 64-bit system, both running Slackware 14.1. The x86_64 system compiles things fine, while the i686 system cannot even build Hello World.

Related issues for this exact exception: #38177 (potentially interesting discussion), #38891 (reported a couple hours before this issue - new regression?)

```
$ cat test.rs
fn main() {
    println!("Hello World!");
}
```

---

```
$ rustup show
Default host: i686-unknown-linux-gnu
...

active toolchain
----------------

nightly-2016-12-16-i686-unknown-linux-gnu (directory override for '/home/i336')
rustc 1.15.0-nightly (8f02c429a 2016-12-15)


$ rustc --version --verbose
rustc 1.15.0-nightly (8f02c429a 2016-12-15)
binary: rustc
commit-hash: 8f02c429ad3e2ad687a222d1daae2e04bb9bb876
commit-date: 2016-12-15
host: i686-unknown-linux-gnu
release: 1.15.0-nightly
LLVM version: 3.9
```

---

```
$ rustc -v test.rs
error: internal compiler error: unexpected panic
...
thread 'rustc' panicked at 'assertion failed: eps.windows(2).all(|w| w[0].cmp(self, &w[1]) != Ordering::Greater)', /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/context.rs:1525
stack backtrace:
   1: 0xb74df235 - std::sys::imp::backtrace::tracing::imp::write::h2b860f17b6b0f511
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:42
   2: 0xb74eb5db - std::panicking::default_hook::{{closure}}::hedea7d569eb1e6b8
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:349
   3: 0xb74eb147 - std::panicking::default_hook::he87e0c76c89c52a4
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:359
   4: 0xb74eba99 - std::panicking::rust_panic_with_hook::h304127bde2d3d5e2
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:553
   5: 0xb66da869 - std::panicking::begin_panic::h1dc990fced681ffe
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:515
   6: 0xb68e2bd5 - rustc::ty::context::TyCtxt::intern_existential_predicates::h20cec3659e8eca6d
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/context.rs:1524
   7: 0xb6d84b5c - <core::result::Result<T, E> as rustc::ty::context::InternIteratorElement<T, R>>::intern_with::hac3e67983ea6f69f
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/context.rs:1559
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/context.rs:1621
   8: 0xb6de70df - <rustc_metadata::decoder::DecodeContext<'a, 'tcx> as serialize::serialize::SpecializedDecoder<&'tcx rustc::ty::Slice<rustc::ty::sty::ExistentialPredicate<'tcx>>>>::specialized_decode::hc3a849731e56212a
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/context.rs:1602
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/context.rs:1559
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_metadata/decoder.rs:436
   9: 0xb6de51ce - <rustc_metadata::decoder::DecodeContext<'a, 'tcx> as serialize::serialize::SpecializedDecoder<&'tcx rustc::ty::TyS<'tcx>>>::specialized_decode::h7e91fc6df8b90ee5
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:746
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:201
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:486
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:195
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:485
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:175
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:150
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:171
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:92
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:164
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:92
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_metadata/decoder.rs:374
  10: 0xb6de533f - <rustc_metadata::decoder::DecodeContext<'a, 'tcx> as serialize::serialize::SpecializedDecoder<&'tcx rustc::ty::TyS<'tcx>>>::specialized_decode::h7e91fc6df8b90ee5
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:746
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:175
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:122
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:171
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:92
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:164
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:92
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_metadata/decoder.rs:374
  11: 0xb6d85e63 - <<core::result::Result<V, E> as core::iter::traits::FromIterator<core::result::Result<A, E>>>::from_iter::Adapter<Iter, E> as core::iter::iterator::Iterator>::next::h0a8d64550acffb73
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:746
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_metadata/decoder.rs:415
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/ops.rs:2645
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/option.rs:383
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/iter/mod.rs:999
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/result.rs:1049
  12: 0xb6d85914 - <core::result::Result<V, E> as core::iter::traits::FromIterator<core::result::Result<A, E>>>::from_iter::hd350930983eccfcd
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/iter/iterator.rs:2180
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_data_structures/array_vec.rs:150
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_data_structures/accumulate_vec.rs:100
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/result.rs:1066
  13: 0xb6d84c29 - <core::result::Result<T, E> as rustc::ty::context::InternIteratorElement<T, R>>::intern_with::hbaf6c08a327678b0
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/iter/iterator.rs:1218
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/context.rs:1621
  14: 0xb6de6dcf - <rustc_metadata::decoder::DecodeContext<'a, 'tcx> as serialize::serialize::SpecializedDecoder<&'tcx rustc::ty::Slice<&'tcx rustc::ty::TyS<'tcx>>>>::specialized_decode::h53b0fc8efae5e796
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/context.rs:1602
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/context.rs:1564
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_metadata/decoder.rs:415
  15: 0xb6db6cd5 - <rustc::ty::sty::FnSig<'tcx> as serialize::serialize::Decodable>::decode::{{closure}}::hf5f247ddd0b104e2
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:746
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:201
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:566
  16: 0xb6db7b80 - <rustc::ty::sty::BareFnTy<'tcx> as serialize::serialize::Decodable>::decode::{{closure}}::h66183a2032b733f8
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:195
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:564
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:201
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:486
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:195
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:485
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:201
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:546
  17: 0xb6de6e4f - <rustc_metadata::decoder::DecodeContext<'a, 'tcx> as serialize::serialize::SpecializedDecoder<&'tcx rustc::ty::sty::BareFnTy<'tcx>>>::specialized_decode::h1196ca5f3694d105
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:195
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:542
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_metadata/decoder.rs:421
  18: 0xb6de5d32 - <rustc_metadata::decoder::DecodeContext<'a, 'tcx> as serialize::serialize::SpecializedDecoder<&'tcx rustc::ty::TyS<'tcx>>>::specialized_decode::h7e91fc6df8b90ee5
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:746
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:175
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:142
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:171
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:92
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:164
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/sty.rs:92
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_metadata/decoder.rs:374
  19: 0xb6deb078 - rustc_metadata::decoder::<impl rustc_metadata::cstore::CrateMetadata>::get_type::h31d5d3fe2ee006e5
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libserialize/serialize.rs:746
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_metadata/decoder.rs:156
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_metadata/decoder.rs:673
  20: 0xb6df200d - rustc_metadata::cstore_impl::<impl rustc::middle::cstore::CrateStore<'tcx> for rustc_metadata::cstore::CStore>::item_type::h076521db7b251ac3
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_metadata/cstore_impl.rs:88
  21: 0xb68f5f38 - rustc::ty::<impl rustc::ty::context::TyCtxt<'a, 'gcx, 'tcx>>::item_type::h7dd05669dfd5ee60
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/mod.rs:2315
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/mod.rs:1903
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/dep_graph/dep_tracking_map.rs:145
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/mod.rs:1899
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/mod.rs:2313
  22: 0xb70d25d9 - rustc_trans::back::symbol_names::<impl rustc_trans::monomorphize::Instance<'tcx>>::symbol_name::h49e3929e31f9c552
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_trans/back/symbol_names.rs:234
  23: 0xb70d115a - rustc_trans::back::symbol_export::ExportedSymbols::compute_from::h4bf627b7435e6774
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_trans/back/symbol_export.rs:95
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/ops.rs:2645
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/option.rs:383
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/iter/mod.rs:999
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcollections/vec.rs:1632
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcollections/vec.rs:1615
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcollections/vec.rs:1502
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/iter/iterator.rs:1218
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_trans/back/symbol_export.rs:89
  24: 0xb70f5e84 - rustc_trans::base::trans_crate::hed2e3c9ed492d3c3
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_trans/base.rs:1733
  25: 0xb767a3ff - rustc_driver::driver::phase_4_translate_to_llvm::h655a38398d71b176
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_driver/driver.rs:1046
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/util/common.rs:34
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_driver/driver.rs:1044
  26: 0xb7648357 - rustc_driver::driver::compile_input::{{closure}}::hda32c37b28fba5cf
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_driver/driver.rs:205
  27: 0xb7678922 - rustc_driver::driver::phase_3_run_analysis_passes::{{closure}}::hcabdc0c365c3bebe
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_driver/driver.rs:992
  28: 0xb76606e4 - rustc_driver::driver::phase_3_run_analysis_passes::hf8046635ec9be751
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/context.rs:1033
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/thread/local.rs:245
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/context.rs:1030
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/context.rs:1017
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/thread/local.rs:245
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/context.rs:1014
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc/ty/context.rs:782
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_driver/driver.rs:866
  29: 0xb7646400 - rustc_driver::driver::compile_input::h04b031728fb79cbb
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_driver/driver.rs:172
  30: 0xb768cb7d - rustc_driver::run_compiler::h458a3cb95357633f
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_driver/lib.rs:221
  31: 0xb75ae0d8 - std::panicking::try::do_call::h9ad582c648ba2206
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_driver/lib.rs:1117
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_driver/lib.rs:137
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_driver/lib.rs:1051
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panic.rs:295
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:458
  32: 0xb74f5212 - __rust_maybe_catch_panic
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libpanic_unwind/lib.rs:98
  33: 0xb75cf37b - <F as alloc::boxed::FnBox<A>>::call_box::he04301a385ee17ba
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panicking.rs:434
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/panic.rs:351
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/thread/mod.rs:287
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/liballoc/boxed.rs:605
  34: 0xb74ea40c - std::sys::imp::thread::Thread::new::thread_start::h460a0adc61139b06
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/liballoc/boxed.rs:615
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sys_common/thread.rs:21
                at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sys/unix/thread.rs:84
  35: 0xb313b678 - start_thread
  36: 0xb73b958d - clone
  37:        0x0 - <unknown>
```

**Comment by mbrubeck** [comment]
This looks like a duplicate of #38177 which is fixed in the latest nightly by #38357.

**Comment by i336** [comment]
Ah, I see. Thanks very much. Closing since that seems appropriate, if anyone has any further comment feel free to add it.
